ui/admin/app/components/form/worker/create-tags/index.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { A } from '@ember/array';
+
+class Tag {
+  @tracked key;
+  @tracked value;
+
+  constructor(key, value) {
+    this.key = key;
+    this.value = value;
+  }
+}
+
+export default class FormWorkerCreateTagsIndexComponent extends Component {
+  // =services
+
+  @service router;
+  @service confirm;
+  @service intl;
+
+  // =attributes
+
+  @tracked apiTags = A([]);
+  hasRemovedExitHandler = false;
+
+  constructor() {
+    super(...arguments);
+    this.addExitHandler();
+  }
+
+  addExitHandler() {
+    this.router.on('routeWillChange', this, this.confirmTransition);
+  }
+
+  removeExitHandler() {
+    if (!this.hasRemovedExitHandler) {
+      this.router.off('routeWillChange', this, this.confirmTransition);
+      this.hasRemovedExitHandler = true;
+    }
+  }
+
+  confirmTransition(transition) {
+    if (transition.isAborted) return;
+
+    if (this.apiTags.length) {
+      this.confirmExit(transition);
+    }
+  }
+
+  async confirmExit(transition) {
+    transition.abort();
+    try {
+      await this.confirm.confirm(this.intl.t('questions.abandon-confirm'), {
+        title: 'titles.abandon-confirm',
+        confirm: 'actions.discard',
+      });
+      this.apiTags = A([]);
```

Comment:
question for understanding: why do we need to set back to empty array before retrying the transition ?